packages/features/apps/AdminAppsList.tsx

```diff
@@ -232,22 +235,43 @@ const EditKeysModal: FC<{
                 key={key}
                 control={formMethods.control}
                 defaultValue={keys && keys[key] ? keys?.[key] : ""}
-                render={({ field: { value } }) => (
-                  <TextField
-                    label={key}
-                    key={key}
-                    name={key}
-                    value={value}
-                    onChange={(e) => {
-                      formMethods.setValue(key, e?.target.value);
-                    }}
-                  />
-                )}
+                render={({ field: { value } }) => {
+                  const isPasswordField =
+                    key.toLowerCase().includes("secret") ||
+                    key.toLowerCase().includes("password") ||
+                    key.toLowerCase().includes("key");
+                  const FieldComponent = isPasswordField ? PasswordField : TextField;
+
+                  return (
+                    <FieldComponent
+                      label={key}
+                      key={key}
+                      name={key}
+                      value={value}
+                      containerClassName="pb-5"
+                      hint={
+                        (appKeySchema.shape as Record<string, { description?: string } | undefined>)[key]
+                          ?.description
+                      }
+                      onChange={(e) => {
+                        formMethods.setValue(key, e?.target.value);
+                      }}
+                    />
+                  );
+                }}
```

Comment:
Hey, as I mentioned, only an admin can open this modal and set values, so I don't think any changes are needed here

Reply:
Which changes are you referring to exactly?

Using password fields for sensitive data is a general measurement to increase data security. No matter if admin or not there's always the possibility that someone's looking at their screen.

Adding the hints was especially necessary for the Jitsi and Nextcloud Talk path patterns. Jitsi doesn't even have a README file. When I was using the Jitsi app in Cal.com for the first time I had to look into the code to understand how that field is working. I think that's not feasible for a non-dev admin. So, I'd rather keep those hints. Or do you have another suggestion to document the usage of those fields?

Reply:
No, I don't think attaching a documentation URL could resolve user confusion. There isn’t a specific URL that provides a step-by-step guide for retrieving credentials and similar tasks. That's why we always ask for a well-written README. Could you pls add A Readme for Jitsi